ucm2/Intel/sof-hda-dsp/HiFi-sof.conf

```diff
@@ -0,0 +1,117 @@
+		HeadphoneIirBlob "${var:BlobPath}/eq_iir/pass.blob"
+		HeadphoneFirBlob "${var:BlobPath}/eq_fir/pass.blob"
+		HeadphoneDrcBlob "${var:BlobPath}/drc/passthrough.blob"
+	}
+}
+
+Define.ConfPathFromDMI "${sys:devices/virtual/dmi/id/sys_vendor}/${sys:devices/virtual/dmi/id/product_name}.conf"
+Define.SOFProductConfig "/blobs/sof/product_configs/${var:ConfPathFromDMI}"
+Define.SOFConfFullPath "${ConfTopDir}${var:SOFProductConfig}"
+Define.SOFUserConfig "/blobs/sof/user_configs/${var:ConfPathFromDMI}"
+Define.SOFUserConfFullPath "${ConfTopDir}${var:SOFUserConfig}"
+
+If.HasUserConfig {
+	Condition {
+		Type Path
+		Mode read
+		Path "${var:SOFUserConfFullPath}"
+	}
+	True {
+		Include.product-config.File "${var:SOFUserConfig}"
+	}
+	False.If.HasProductConfig {
+		Condition {
+			Type Path
+			Mode read
+			Path "${var:SOFConfFullPath}"
+		}
+		True {
+			Include.product-config.File "${var:SOFProductConfig}"
+		}
+	}
+}
+
+# Redefine headphone control FOR SOF. This is same as Headphone in HDA/HiFi-analog.conf
+# but adds DRC and EQ controls.
+
+DefineMacro.SofControl {
+	Define.EndpointDrcVar "${var:__endpoint}DrcBlob"
+	Define.EndpointIirVar "${var:__endpoint}IirBlob"
+	Define.EndpointFirVar "${var:__endpoint}FirBlob"
+	Define.EndpointDrcBlob "${var:$EndpointDrcVar}"
+	Define.EndpointIirBlob "${var:$EndpointIirVar}"
+	Define.EndpointFirBlob "${var:$EndpointFirVar}"
+
+       If.endpoint_with_drc {
+		Condition {
+			Type ControlExists
+			Control "name='${var:PostMixerAnalogPlaybackDRCSwitch}'"
+		}
+		True {
+			EnableSequence [
+				# For debug uncomment below the echo command to see what blobs are applied
+				#shell "/bin/echo '${var:__endpoint} ${var:EndpointIirBlob} ${var:EndpointFirBlob} ${var:EndpointDrcBlob}' >> /tmp/alsa-ucm.txt"
+				cset "name='${var:PostMixerAnalogPlaybackDRCSwitch}' ${var:__drcswitch}"
+				cset-tlv "name='${var:PostMixerAnalogPlaybackDRCBytes}' ${var:EndpointDrcBlob}"
+				cset-tlv "name='${var:PostMixerAnalogPlaybackIIRBytes}' ${var:EndpointIirBlob}"
+				cset-tlv "name='${var:PostMixerAnalogPlaybackFIRBytes}' ${var:EndpointFirBlob}"
+			]
+		}
+		False.If.endpoint_with_eq {
+			Condition {
+				Type ControlExists
+				Control "name='${var:PostMixerAnalogPlaybackFIRBytes}'"
+			}
+			True {
+				EnableSequence [
+					# For debug uncomment below the echo command to see what blobs are applied
+					#shell "/bin/echo '${var:__endpoint} ${var:EndpointIirBlob} ${var:EndpointFirBlob}' >> /tmp/alsa-ucm.txt"
+					cset-tlv "name='${var:PostMixerAnalogPlaybackIIRBytes}' ${var:EndpointIirBlob}"
+					cset-tlv "name='${var:PostMixerAnalogPlaybackFIRBytes}' ${var:EndpointFirBlob}"
+				]
+			}
+		}
+	}
+}
+
+# Merge this add to Headpones subtree in HDA/HiFi-analog.conf
+SectionDevice."Headphones" {
+	Macro.headphone.SofControl "endpoint=Headphone drcswitch=off"
+}
+
+# Merge this to Speaker subtree in HDA/HiFi-analog.conf
+SectionDevice."Speaker" {
+	Macro.speaker.SofControl "endpoint=Speaker drcswitch=on"
+}
```

Comment:
Will this create a "Speaker" Section Device when there is no Speaker, like on upx-i11?

Reply:
Yes, the device will be always defined in this case. Move those lines to `If` block to use it only when the `Speaker Playback Switch` control exists.